config/config.sample.yaml

```diff
@@ -1,25 +1,67 @@
+# Port to listen on for incoming slack requests from webhooks and event API messages.
+# Both options are optional if using RTM, required otherwise.
 slack_hook_port: 9898
+# Prefix of incoming requests to strip when handing. This is NOT the bind host
 inbound_uri_prefix: "https://my.server.here:9898/"
-bot_username: "slackbot"
-username_prefix: "slack_"
 
-homeserver:
-  url: http://localhost:8008
+# TLS files to be used when handing slack requests
+# Optional
+tls:
+  key_file: /path/to/tls.key
+  crt_file: /path/to/tls.crt
+
+# Optional if slack_hook_port and inbound_uri_prefix are defined, required otherwise.
+rtm:
+  # Use the RTM API to listen for requests, which does not require
+  # the bridge to listen on a port. This should be on unless you have
+  # specific needs.
+  enable: true
+  logging: "silent" # Logging level specific to RTM traffic.
+
+# Allow users to add channels dynamically by using oauth, or puppet themselves.
+# Optional
+oauth2:
+  client_id: ""
+  client_secret: ""
+  # A prefix similar to inbound_uri_prefix for oauth2 requests. inbound_uri_prefix will be used if this is not set
+  # Optional
+  redirect_prefix: ""
+
+# The prefix to give slack users 
+username_prefix: "slack_" # Required
+
+homeserver: # Required
+  # The server_name of your homeserver
   server_name: my.server
+  # The URL which the bridge can reach the homeserver on.
+  url: http://localhost:8008
+  # The URL which slack can reach media on.
+  media_url: "http://my.server.here"
 
+# Optional
 logging:
-  console: "info"
+  # Log level for console output
```

Comment:
```suggestion
  # The default log level for the bridge
```

Reply:
Huh, but it's not the default? It's the actual level at which logs will be outputted to the console/stdout?

Reply:
Which would be everything that's unhandled, ergo default. The comment doesn't add anything otherwise, so I would recommend changing it or removing it.

Reply:
Probably better to remove it.